tests/unit/test_charm.py

```diff
@@ -372,59 +548,72 @@ def test_given_csr_stored_when_relation_joined_then_csr_not_generated_again(
             relation_id=relation_id, remote_unit_name="certificates-provider/0"
         )
 
-        self.assertEqual(self.harness._backend.secret_get(label="csr-0")["csr"], CSR)
+        secret = self.harness.model.get_secret(label="csr-0")
+        secret_content = secret.get_content(refresh=True)
+        self.assertEqual(secret_content["csr"], self.csr.decode())
         patch_generate_csr.assert_not_called()
 
+    @patch("charms.tls_certificates_interface.v3.tls_certificates.TLSCertificatesRequiresV3.get_certificate_signing_requests"  # noqa: E501, W505)
+    )
     @patch(
         "charms.tls_certificates_interface.v3.tls_certificates.TLSCertificatesRequiresV3.request_certificate_creation"  # noqa: E501, W505
     )
-    def test_given_certificate_stored_when_relation_joined_then_certificate_not_requested_again(
+    def test_given_certificate_requested_when_configure_then_certificate_not_requested_again(
```

Comment:
```suggestion
    def test_given_certificate_requested_when_configure_and_configuration_has_not_changed_then_certificate_not_requested_again(
```
The name I proposed might not be the best but the idea is to make it clear that only if the config has not changed (for example the common name) the certificate is not requested again.

Reply:
Similarly in `test_given_csr_stored_when_relation_joined_then_csr_not_generated_again`

Reply:
I'm not sure I agree with the recommendation, the "when"/event is the `configure`, not `configure_and_configuration_has_not_changed`. The following seems correct to me:

- Given: Certificate already requested
- When: Configure
- Then: Certificate not requested again

Which is what is already there.

Reply:
If I was to read this before reading the charm code it would give me the impression that a new certificate should not be requested if we already requested one. That is true but not always. 
And in the case of `test_given_csr_stored_when_relation_joined_then_csr_not_generated_again` it's fine since we now have the `test_given_csr_stored_when_config_changed_then_new_certificate_is_requested` test case.
I don't know if having a `test_given_certificate_requested_when_config_changed_then_new_certificate_is_requested` would be an overkill.
But this is where I'm coming from. I'll leave this to your judgement and approve the PR as the rest looks good :)